src/spot_skills/scripts/spot_arm_action_server.py

```diff
@@ -3,17 +3,25 @@
 """Provide an action server to control Spot's arm using a joint trajectory."""
 
 import rospy
+import std_msgs.msg
 
 from spot_skills.follow_trajectory_server import SpotFollowJointTrajectoryServer
-from spot_skills.ros_utilities import get_ros_params
+from spot_skills.ros_utilities import BoolCallbackWrapper, get_ros_params
 from spot_skills.spot_arm_controller import SpotArmController
 from spot_skills.spot_manager import SpotManager
 
 
 def main() -> None:
-    """Create a ROS node for the action server and then spin."""
+    """Create a ROS node for the action server once Spot is ready, then spin."""
     rospy.init_node("spot_follow_joint_trajectory")
 
+    # Wait for the signal that Spot is ready to be controlled
+    wait_for_bool = BoolCallbackWrapper("spot/robot_ready")
+    loop_rate = rospy.Rate(5)  # Loop at 5 Hz
+
+    while not (rospy.is_shutdown() or wait_for_bool.received_true):
+        loop_rate.sleep()
```

Comment:
I think this needs a `spin_once` call.

Reply:
Although `rospy` doesn't include `spinonce`, I refactored the server to use two `rospy.spin`, one within a callback.